lib/errors.ts

```diff
@@ -0,0 +1,61 @@
+import ExtendableError from 'es6-error';
+import { NextResponse } from 'next/server';
+
+type ServerErrorParams = {
+  error_message: string;
+  error_type: string;
+  status_code: number;
+  underlyingError?: unknown;
+};
+
+export class ServerError extends ExtendableError {
+  params: ServerErrorParams;
+  trace_id?: string;
+
+  constructor(params: ServerErrorParams) {
+    super(params.error_type);
+    this.params = params;
+  }
+
+  toJSON() {
+    return {
+      error_id: this.trace_id,
```

Comment:
this will be `sentry id` or something when we do https://linear.app/experiment-station/issue/SUPA-15/monitoring-error-tracking